lib/src/dart_formatter.dart

```diff
@@ -200,17 +210,9 @@ class DartFormatter {
   // happens to parse without error, then we use that result instead.
   ParseStringResult _parse(String source, String? uri,
       {required bool patterns}) {
-    // Enable all features that are enabled by default in the current analyzer
-    // version.
-    FeatureSet featureSet;
-    if (patterns) {
-      featureSet = FeatureSet.latestLanguageVersion();
-    } else {
-      featureSet = FeatureSet.fromEnableFlags2(
-        sdkLanguageVersion: Version(2, 19, 0),
-        flags: [],
-      );
-    }
+    var version = patterns ? Version(3, 0, 0) : Version(2, 19, 0);
```

Comment:
```suggestion
    var version = patterns ? FeatureSet.latestLanguageVersion() : Version(2, 19, 0);
```

Reply:
That doesn't work since `FeatureSet.latestLanguageVersion()` is a `FeatureSet` and not a `Version`. I can't just use that `FeatureSet` directly either, because then it doesn't accept a list of experiment flags.

I'll just stick with hardcoding the version. This should work fine because whenever new syntactic features are added, I have to update the formatter anyway and will update the version then.

Reply:
I don't know that this will interact well with language experiments still? Maybe it is fine, I don't know for sure, but it definitely doesn't feel right to hardcode it?